test/plain.jl

```diff
@@ -1055,6 +1055,67 @@ dset = HDF5.create_external_dataset(hfile, "ext", fn_external, Int, (10,20))
 
 end
 
-# length for FixedString
-fix = HDF5.FixedString{4,0}((b"test"...,))
-@test length(fix) == 4
+@testset "FixedStrings and FixedArrays" begin
+    # properties for FixedString
+    fix = HDF5.FixedString{4,0}((b"test"...,))
+    @test length(typeof(fix)) == 4
+    @test length(fix) == 4
+    @test HDF5.pad(typeof(fix)) == 0
+    @test HDF5.pad(fix) == 0
+    # issue #742, large fixed strings are readable
+    mktemp() do path, io
+        close(io)
+        ref = join('a':'z') ^ 1000
+        fid = h5open(path, "w")
+        # long string serialized as FixedString
+        fid["longstring"] = ref
+
+        # compound datatype containing a FixedString
+        compound_dtype = HDF5.Datatype(HDF5.h5t_create(HDF5.H5T_COMPOUND, sizeof(Int64) + sizeof(ref)))
+        HDF5.h5t_insert(compound_dtype, "n", 0, datatype(Int64))
+        HDF5.h5t_insert(compound_dtype, "a", sizeof(Int64), datatype(ref))
+        c = create_dataset(fid, "compoundlongstring", compound_dtype, dataspace(()))
+        # normally this is done with a `struct name; n::Int64; a::NTuple{N,Char}; end`, but
+        # we need to not actually instantiate the NTuple.
```

Comment:
What's the rationale for this. Is it to avoid instatiating very large NTuple, which would break  without this patch.

Reply:
Yes, it's to avoid instantiating large tuples, but at the Julia level, independent from trying to work around the issue here. Actually instantiating the example `name` struct would run afoul of e.g. https://github.com/JuliaLang/julia/issues/35619#issuecomment-731558911